assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/index.test.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * SearchFunnelWidget component tests.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+	render,
+	createTestRegistry,
+	provideModules,
+	provideUserCapabilities,
+	waitForElementToBeRemoved,
+	provideUserInfo,
+	provideUserAuthentication,
+	provideSiteInfo,
+} from '../../../../../../../tests/js/test-utils';
+import coreModulesFixture from '../../../../../googlesitekit/modules/datastore/__fixtures__';
+import { CORE_MODULES } from '../../../../../googlesitekit/modules/datastore/constants';
+import { CORE_USER } from '../../../../../googlesitekit/datastore/user/constants';
+import { MODULES_SEARCH_CONSOLE } from '../../../datastore/constants';
+import { getWidgetComponentProps } from '../../../../../googlesitekit/widgets/util';
+import SearchFunnelWidget from '.';
+
+describe( 'SearchFunnelWidget', () => {
+	let registry;
+
+	const widgetComponentProps = getWidgetComponentProps( 'searchFunnel' );
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+
+		provideModules( registry );
+		provideUserInfo( registry );
+		provideUserAuthentication( registry );
+		provideUserCapabilities( registry );
+		provideSiteInfo( registry );
+		registry.dispatch( CORE_USER ).receiveConnectURL( 'test-url' );
+		registry.dispatch( MODULES_SEARCH_CONSOLE ).receiveGetSettings( {
+			propertyID: 'http://example.com/',
+		} );
+		registry.dispatch( CORE_USER ).setReferenceDate( '2021-01-28' );
+
+		fetchMock.getOnce(
+			/^\/google-site-kit\/v1\/modules\/search-console\/data\/searchanalytics/,
+			{
+				body: [
+					{
+						clicks: 123,
+						ctr: 4.56,
+						impressions: 7890,
+						keys: [ '2022-06-21' ],
+						position: 12.345,
+					},
+				],
+			}
+		);
+	} );
+
+	it( 'should render the Search Funnel Widget, including the Activate Analytics CTA', async () => {
+		const { container, getByText } = render(
+			<SearchFunnelWidget { ...widgetComponentProps } />,
+			{ registry }
+		);
+
+		await waitForElementToBeRemoved(
```

Comment:
Is it not possible to use `waitForRegistry` here instead? If so, it would probably be worth leaving a comment to explain 👍

Reply:
Thanks for the suggestion - `waitForRegistry` works here, no need to wait for a specific element to be removed. I've made the change here and elsewhere in the PR :+1: